### DISCUSSION TITLE
pch.h vs. stdafx.h: why two different precompiled headers?

### DISCUSSION THREAD

**Discussion by joce** [WinMerge/winmerge#2020] [Q&A]
I was looking at the code, and I can't help but wonder why are there two different precompiled headers being used in the project. Does anyone know?

**Comment by sdottaka** [comment]
stdafx.h contains header files for the MFC library, and pch.h contains header files that are independent of the MFC library and Windows.
In order to improve portability, we are trying to reduce dependence on the MFC library, so source files that do not use MFC include pch.h.

**Reply by joce** [reply]
I see. Thanks for the information! 😄 

However, given there is no distinction for the files that require MFC and those who don't (meaning the UI/MFC relying part isn't a separate library), it leads to a weird state where you have these. I wonder if there's an effect on the compilation time since you can't specify both precompiled headers at the project level:

![image](https://github.com/WinMerge/winmerge/assets/80361/7d97a77e-eea4-4c89-b40f-fbdc44a817dd)

**Comment by sdottaka** [comment]
It is acceptable to increase compilation time by separating precompiled headers.
Note that WinMergeU.pch is created from stdafx.h, and WinMergU2.pch is created from pch.h.
